### PR TITLE
add return to param CVR-774

### PR DESCRIPTION
**Pull request by hobbitronics**
### Added

- redirect to return-to param

[CVR-774](https://itse.youtrack.cloud/issue/CVR-774)

--
### Feature branch checklist

- [ ] Documentation (README, etc.)
- [ ] Run `make format`, `make lint` and `make check`

### CODE REVIEW

File: src/authn/index.ts [hunk 1]
```diff
@@ -10,10 +11,14 @@ export type AuthLoginResponse = {
 export const showApp = writable(false)
 
 export const login = async (inviteCode = ''): Promise<void> => {
+  const returnUrl = location.pathname
   let url = 'auth/login'
   if (inviteCode) {
     url += `&invite=${inviteCode}`
```
Comment (briskt) [inline src/authn/index.ts:17]: Does this work? Seems like it should be `?invite=`

File: src/authn/index.ts [hunk 2]
```diff
@@ -10,10 +11,14 @@ export type AuthLoginResponse = {
...
   }
+  if (returnUrl !== ROOT && returnUrl !== HOME) {
+    url += `?return-to=${returnUrl}`
```
Comment (briskt) [inline src/authn/index.ts:20]: This may need to be conditional, based on whether or not there's already a `?` in the URL.

**Review by briskt — CHANGES_REQUESTED** [review]

File: src/authn/index.ts (hunk 1 above)
Comment (hobbitronics) [reply src/authn/index.ts:17]: I haven't tested it, but wondered about that.

Comment (hobbitronics) [reply src/authn/index.ts]: I'm not sure how to get an invitation code to test

**Review by briskt — APPROVED** [review]

File: src/authn/index.ts [hunk 3]
```diff
@@ -10,9 +11,13 @@ export type AuthLoginResponse = {
 export const showApp = writable(false)
 
 export const login = async (inviteCode = ''): Promise<void> => {
+  const returnUrl = location.pathname
   let url = 'auth/login'
   if (inviteCode) {
-    url += `&invite=${inviteCode}`
+    url += `?invite=${inviteCode}`
+  }
+  if (returnUrl !== ROOT && returnUrl !== HOME) {
```
Comment (hobbitronics) [inline src/authn/index.ts:19]:
```suggestion
  if (returnUrl !== ROOT && returnUrl !== HOME && !inviteCode) {
```

File: src/authn/index.ts [hunk 4]
```diff
@@ -10,9 +11,13 @@ export type AuthLoginResponse = {
...
+    url += returnUrl.includes('?') || inviteCode ? `&return-to=${returnUrl}` : `?return-to=${returnUrl}`
```
Comment (hobbitronics) [inline src/authn/index.ts:20]:
```suggestion
    url += returnUrl.includes('?') ? `&return-to=${returnUrl}` : `?return-to=${returnUrl}`

```

**Review by hobbitronics — COMMENTED** [review]
Maybe we should just not include the return-to when logging in from an invite code url.

File: src/authn/index.ts (hunk 3 above)
Comment (briskt) [reply src/authn/index.ts:19]: Yeah, probably not.

**Pull request merged**